agent/src/softwarecontaineragent.cpp

```diff
@@ -152,50 +154,31 @@ ContainerID SoftwareContainerAgent::findSuitableId()
     return availableID;
 }
 
-SoftwareContainerAgent::SoftwareContainerPtr SoftwareContainerAgent::makeSoftwareContainer(const ContainerID containerID)
-{
-    // At this point, the configs are all gathered and should be unique to each SC instance
-    // so we create a copy and pass along ownership.
-    std::unique_ptr<SoftwareContainerConfig> config =
-        std::unique_ptr<SoftwareContainerConfig>(new SoftwareContainerConfig(m_containerConfig));
-
-    auto container = SoftwareContainerPtr(new SoftwareContainer(containerID,
-                                                                std::move(config)));
-    log_debug() << "Created container with ID :" << containerID;
-
-    return container;
-}
-
 ContainerID SoftwareContainerAgent::createContainer(const std::string &config)
 {
     profilepoint("createContainerStart");
     profilefunction("createContainerFunction");
 
-    ContainerConfigParser::ContainerOptions options = m_configParser.parse(config);
-    m_containerConfig.setEnableWriteBuffer(options.enableWriteBuffer);
-
-    std::pair<ContainerID, SoftwareContainerPtr> pair = std::move(getContainerPair());
+    // Set options for this container
+    DynamicContainerOptions options = m_configParser.parse(config);
```

Comment:
This section of the code makes me feel the naming of the members (especially), and the parameter etc. could be more separated between "config" and "options". Maybe rename m_configParser?